service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.domain.dto.UserDTO;
+import com.epam.esm.domain.entity.User;
+import com.epam.esm.service.UserService;
+import com.epam.esm.service.exception.IdNotExistException;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class UserServiceImpl implements UserService {
+
+    UserDao userDao;
+    ModelMapper modelMapper;
+
+    public UserServiceImpl(UserDao userDao, ModelMapper modelMapper) {
+        this.userDao = userDao;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public UserDTO readById(Long id) {
+        Optional<User> user = userDao.readById(id);
+        if (user.isPresent()) {
+            return modelMapper.map(user.get(), UserDTO.class);
+        } else {
+            throw new IdNotExistException(id.toString());
+        }

```

Comment:
.map(...)
.orElseThrow(...)

could be used

Reply:
thanks for advice. replaced